Melanzana.MachO/MachRelocation.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Melanzana.MachO
+{
+    public class MachRelocation
+    {
+        /// <summary>
+        /// Address from the start of the section.
+        /// </summary>
+        public int Address { get; init; }
+
+        /// <summary>
+        /// Symbol index (for internal symbols) or section index (for external symbols).
+        /// </summary>
+        public uint SymbolOrSectionIndex { get; init; }
```

Comment:
TODO: Use symbolic reference here to `MachSection` or `MachSymbol`.

Reply:
I am bit torn on this since both approaches are somewhat broken at the moment.

For reference, LibObjectFile uses index in `ElfRelocation` and object reference to section in `ElfSymbol`. The current approach mirror that but we lack indexing into the symbol table at the moment.

Reply:
While not perfect it seems that this API shape works for generating quite complex object files. I tested it to power NativeAOT compiler and compared the output to LLVM generated one.